internal/cmd/environment/command_update.go

```diff
@@ -34,9 +36,14 @@ func (c *command) update(cmd *cobra.Command, args []string) error {
 	}
 
 	environment := orgv2.OrgV2Environment{DisplayName: orgv2.PtrString(name)}
-	environment, err = c.V2Client.UpdateOrgEnvironment(args[0], environment)
-	if err != nil {
-		return err
+	if environment, err = c.V2Client.UpdateOrgEnvironment(args[0], environment); err != nil {
+		environmentId, err := nameconversions.EnvironmentNameToId(args[0], c.V2Client, false)
+		if err != nil {
+			return errors.NewErrorWithSuggestions(err.Error(), errors.NotValidEnvironmentIdSuggestions)
+		}
+		if environment, err = c.V2Client.UpdateOrgEnvironment(environmentId, environment); err != nil {
```

Comment:
There's a variable name collision here. The `environment` for the request on line 38 gets modified by line 39, and that's what will be sent here.